### DISCUSSION TITLE
Login without GitHub

### DISCUSSION THREAD

**Discussion by Jak2k** [zed-industries/zed#24737] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

I want to log in without my GitHub account. This could be implemented using mail, something like Mastodon OAuth or Indie Login.

### Environment

Zed: v0.164.2 (Zed)
OS: Linux Wayland fedora 41
Memory: 31.1 GiB
Architecture: x86_64
GPU: NVIDIA GeForce GTX 1650 with Max-Q Design || NVIDIA || 565.57.01

**Comment by notpeter** [comment]
Current GitHub is the only authentication provider we support. Perhaps at some point in the future we'll support alternate SSO auth providers (GitLab, Google OAuth, Custom SAML, etc) as there is market demand.

**Reply by Jak2k** [reply]
I'd prefer signing in without relying on a single company. Mail would be the most straightforward solution.

**Reply by jalvarado91** [reply]
Hello, sorry to revive this thread but wondering if there's any change on these plans as paying for zed services is not out of the ordinary. My employer can pay for the cost of Zed Pro, however my github account is tied to my personal email, and so my zed account is tied to my personal email. If there was a way to log in with google for example, then keeping personal/work accounts wouldn't be an issue

**Reply by andreashaerter** [reply]
> I'd prefer signing in without relying on a single company. Mail would be the most straightforward solution.

Exactly. I would like to go Pro (as I am very happy with Zed, replaced VS Code here + looking into the collaboration feature) but I will never ever use "Sign in with Github" and/or another SSO provider. Please please please enable usual account creation with username/email+password and some kind of 2FA.

**Comment by Pshemas** [comment]
for me the way Zed is tied to a single provider is one of the key reasons I'm not using it. Heck, the mere idea of logging in to text editor seems a bit strange to me - why? 
Integration service / plugin that would allow working with Github / Gitlab / Gitea / whatever service you want, if you need, would make so much more sense.

**Reply by Jak2k** [reply]
I guess it's because of their AI focus. If you just want a fast editor/IDE, I can recommend Helix.

**Reply by ksed** [reply]
Wholly concur. Github exclusivity prevents me from ever considering it for professional use since Github and other non-self-hosted platforms are a total no-go.

**Comment by Gerharddc** [comment]
I am using https://github.com/Gerharddc/litterbox to run Zed inside a security sandbox (i.e. a special container) for improved security during development. It works surprisingly well so far but it is a bit of an issue to sign into Zed. When there is a browser installed in the container, pressing "Sign in" opens said browser and prompts for logging into Zed using GitHub. This is where the problem lies since I don't want to have the browser in the Litterbox be logged into my GitHub account for security reasons. Hence, it would be very useful if there were some other way to log into Zed with minimal permissions i.e. through a access token or similar.

**Reply by eriteric** [reply]
You may have picked one of the only editors that is a bad fit for your use case. But, zed should still hear about it imo. Agree you shouldn't have to sign in just to use the thing.

**Comment by guilleliss** [comment]
Standard email/password would be great

**Reply by Jak2k** [reply]
Maybe try [Gram](https://gram.liten.app/) (a Zed fork).

**Reply by guilleliss** [reply]
interesting !

**Comment by Pshemas** [comment]
For me - if the devs feel logging in brings value (and to a degree I can see that) the better option would be to allow generic OIDC login (or SAML, but OIDC would IMO be preferable). With this I could see a much stronger use case in educational environment (tying it to github is in many instances no go).

**Comment by 4oooo4** [comment]
Quick feedback: GitHub-only login is kind of a dealbreaker for me. I get why it’s there, but not everyone wants to tie their IDE access to microslop.

Right now this basically means I can’t use Zed as my main editor, which is a bit sad because everything else looks really promising.

Any chance you could add other login options (email, etc.)?

**Comment by sstoychev** [comment]
My issue with remote SSH was resolved, for which I'm very happy, and I decided to pay for Zed and try it only to find out I can try it only with Github. 
Well, I can't do this. I don't want every registration I make on internet to have my email, its simply unacceptable. 
What are my options? To have second Github just for Zed and jiggle them?

**Comment by phillamb168** [comment]
Would also really like to see something else for auth besides github. We use Entra/AzureAD for all things corporate and I'd really like to be able to leverage that.

**Comment by dhruvil009** [comment]
This is absurd. Not only is no other mechanism not available. GitHub signup fails, because I have my email address set to private. Why do I have to share my information with public. Zed Sucks

**Comment by skorobogatydmitry** [comment]
Up. One of the major issues users can encounter is if their GH account is tied to a corporate email (as well as personal). Being unsure if they can click "next" => use another provider for coding agent.

**Comment by akrea** [comment]
i would love to use it. for security AND privacy reasons i keep logins seperate for all services and tools. so as long there is no zed isolated login - i will not be using zed, although i would like to on linux.

**Comment by christo-olivier** [comment]
Please please please allow an email and password signup. Given the amount of drama GitHub is facing on a weekly basis it is simply impossible to understand how this is the only sign up option.

**Reply by jgarte** [reply]
Yes please

**Comment by trymeouteh** [comment]
Would like to pay for Zed to use AI models but I never signup for accounts that force you to link your other accounts like github or social accounts.

Please add email, password with optional TOTP.

Also please accept cryptocurrency for payment.
